File: sevent_ lesson_intermediate_certifications/Java_ok/Woman.java
import java.time.LocalDate;

public class Woman extends Person
{
    public boolean HasMakeup  = false;
    protected String HelloPhrase = "Здравствуйте, мальчики, я женщина!";

    public Woman(String name)
    {
        super(name);
    }

    public Woman(String name, LocalDate birthday)
    {
        super(name, birthday);
    }
    @Override
    public void SayHello()
    {
        System.out.printf("%s: Прииивет!, я женщина!\n", name);
    }

    public void PutMakeupOn()
    {
        System.out.printf("%s наносит макияж\n", name);
        this.HasMakeup = true;
    }

    public void RemoveMakeup()
    {
        System.out.printf("%s удаляет макияж\n", name);
        this.HasMakeup = false;
    }

    @Override
    protected void TakeCareImlementation()
    {
        System.out.printf("%s кормит ужином и укладывает спать.\n", name);
    }

    public boolean isHasMakeup() {
        return HasMakeup;
    }

    public void setHasMakeup(boolean hasMakeup) {
        HasMakeup = hasMakeup;
    }

    public String getHelloPhrase() {
        return HelloPhrase;
    }

    public void setHelloPhrase(String helloPhrase) {
        HelloPhrase = helloPhrase;
    }

}

File: sevent_ lesson_intermediate_certifications/C_Sharp_ok/ConsoleApp1/Woman.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    internal class Woman : Person, IBabySitter
    {
        public bool HasMakeup { get; private set; } = false;
        protected override string HelloPhrase { get; set; } = "Здравствуйте, мальчики, я женщина!";

        public Woman(string name) : base(name)
        {
        }

        public Woman(string name, DateTime birthday) : base(name, birthday)
        {
        }

        public sealed override void SayHello()
        {
            Console.WriteLine($"{Name}: Прииивет!, я женщина!");
        }

        public void PutMakeupOn()
        {
            Console.WriteLine($"{Name} наносит макияж");
            this.HasMakeup = true;
        }

        public void RemoveMakeup()
        {
            Console.WriteLine($"{Name} удаляет макияж");
            this.HasMakeup = false;
        }

        protected override void TakeCareImlementation()
        {
  
                Console.WriteLine($"{Name} кормит ужином и укладывает спать.");
        }

        void IBabySitter.TakeCare()
        {
            Console.WriteLine($"{Name} сидит с детьми пока родители на работе.");
        }

    }
}
